src/Slices/ServiceInventory/UI/Components/ForceStateAction.tsx

```diff
@@ -0,0 +1,242 @@
+        tooltipContent={
+          isHalted
+            ? words("environment.halt.tooltip")
+            : words("inventory.statustab.actionDisabled")
+        }
+      >
+        <StyledDropdown
+          toggle={
+            <DropdownToggle
+              data-testid={`${id}-force-state-toggle`}
+              onToggle={() => setIsDropdownOpen(!isDropdownOpen)}
+              toggleIndicator={CaretDownIcon}
+            >
+              {words("inventory.statustab.forceState")}
+            </DropdownToggle>
+          }
+          dropdownItems={dropdownItems}
+          isOpen={isDropdownOpen}
+          onSelect={onSelect}
+        />
+      </ActionDisabledTooltip>
+      <ConfirmationModal
+        confirmationText={confirmationText}
+        onSetInstanceState={onSubmit}
+        id={id}
+        targetState={targetState}
+        isModalOpen={isModalOpen}
+        setIsModalOpen={handleModalToggle}
+        setErrorMessage={setStateErrorMessage}
+      />
+    </>
+  );
+};
+
+interface ModalProps {
+  confirmationText: string;
+  id: string;
+  targetState: string;
+  onSetInstanceState: (targetState: string) => Promise<void>;
+  isModalOpen: boolean;
+  setIsModalOpen: (isOpen: boolean) => void;
+  setErrorMessage: (message: string) => void;
+}
+
+const ConfirmationModal: React.FC<ModalProps> = ({
+  confirmationText,
+  onSetInstanceState,
+  isModalOpen,
+  setIsModalOpen,
+  id,
+  targetState,
+}) => {
+  const handleModalToggle = () => {
+    setIsModalOpen(!isModalOpen);
+  };
+  const handleConfirm = async () => {
+    await onSetInstanceState(targetState);
+    handleModalToggle();
+  };
+
+  return (
+    <React.Fragment>
+      <Modal
+        data-testid={`${id}-force-state-modal`}
+        variant={ModalVariant.small}
+        titleIconVariant="danger"
+        title={words("inventory.statustab.forceState.confirmTitle")}
+        isOpen={isModalOpen}
+        onClose={handleModalToggle}
+        actions={[
+          <Button
+            key="confirm"
+            variant="primary"
+            data-testid={`${id}-force-state-modal-confirm`}
+            onClick={handleConfirm}
+          >
+            {words("yes")}
+          </Button>,
+          <Button
+            key="cancel"
+            variant="link"
+            data-testid={`${id}-force-state-modal-cancel`}
+            onClick={handleModalToggle}
+          >
+            {words("no")}
+          </Button>,
+        ]}
+      >
+        <Text>{confirmationText}</Text>
+        <br />
+        <Text>{words("inventory.statustab.forceState.confirmMessage")}</Text>
+        <Text>{words("inventory.statustab.forceState.confirmQuestion")}</Text>
+      </Modal>
+    </React.Fragment>
+  );
+};
+
+interface ToastAlertMessageProps {
+  stateErrorMessage: string;
+  id: string;
+  setStateErrorMessage: React.Dispatch<React.SetStateAction<string>>;
+}
+
+const ToastAlertMessage: React.FC<ToastAlertMessageProps> = ({
```

Comment:
Same for this one, wasn't there already a component ?

Reply:
fixed